### PR TITLE
Fix meta search

### PR DESCRIPTION
**Pull request by asdfryan**
image

### CODE REVIEW

File: puzzles/templates/puzzles_table.html [hunk 1]
```diff
@@ -180,6 +182,16 @@ <h5 class="modal-title" id="metatitle-{{ p.pk }}">Assign Meta</h5>
         return search_str;
     };
 
+    // Search by the title of the puzzle, and the term "META" if it's a
+    // metapuzzle.
+    $.fn.dataTableExt.ofnSearch['name-input'] = function(value) {
```
Comment (erwa) [inline puzzles/templates/puzzles_table.html]: Hmm, don't really understand the comment above relative to the code below. What's "title" and what's "META" in the code?

Comment (asdfryan) [reply puzzles/templates/puzzles_table.html]: Datatable implements pure substring search and filtering based on the search token (by default, the raw HTML as string).

Here I am rewriting the search token to be the title of the puzzle, with "meta" appended to it if it's a meta puzzle.

Comment (asdfryan) [reply puzzles/templates/puzzles_table.html]: I've updated the comment to be a little more clear.

Comment (asdfryan) [reply puzzles/templates/puzzles_table.html]: I've also changed the jquery to be less magical and more sensible (and added the relevant class tags to the html).

Comment (erwa) [reply puzzles/templates/puzzles_table.html]: Ah, thank you for that. Was confused what those 'a' and '.badge' references were.

**Pull request merged**